notifications/notifiers/email.py

```diff
@@ -57,18 +62,17 @@ def send_notification(self, email_notification):
         """
         messages = None
         user = email_notification.user
-        notification_settings = NotificationSettings.objects.get(
-            user=user,
-            notification_type=email_notification.notification_type,
-        )
 
         with utils.mark_as_sent_or_canceled(email_notification) as will_send:
+            if user != self.user:
+                raise Exception("Notification user doesn't match settings user")
```

Comment:
Are there any cases where this exception would get raised or is this just a precaution?

Reply:
This is a precaution against programmer error.